src/Plugins/ComplexCore/src/ComplexCore/Filters/CropImageGeometry.cpp

```diff
@@ -0,0 +1,492 @@
+  auto oldResolution = getCurrentVolumeDataContainerResolutions(data, srcImagePath);
+
+  auto srcImageGeom = data.getDataAs<ImageGeom>(srcImagePath);
+  auto oldOrigin = srcImageGeom->getOrigin();
+
+  std::vector<usize> tDims(3, 0);
+  if(xMax - xMin < 0)
+  {
+    xMax = xMin + 1;
+  }
+  if(yMax - yMin < 0)
+  {
+    yMax = yMin + 1;
+  }
+  if(zMax - zMin < 0)
+  {
+    zMax = zMin + 1;
+  }
+
+  tDims[0] = (xMax - xMin) + 1;
+  tDims[1] = (yMax - yMin) + 1;
+  tDims[2] = (zMax - zMin) + 1;
+
+  IntVec3 newDimensions;
+  newDimensions[0] = tDims[0];
+  newDimensions[1] = tDims[1];
+  newDimensions[2] = tDims[2];
+
+  FloatVec3 newResolution;
+  newResolution = oldResolution;
+
+  std::vector<float32> newOrigin(3);
+  if(shouldUpdateOrigin)
+  {
+    newOrigin[0] = xMin * newResolution[0] + oldOrigin[0];
+    newOrigin[1] = yMin * newResolution[1] + oldOrigin[1];
+    newOrigin[2] = zMin * newResolution[2] + oldOrigin[2];
+  }
+  else
+  {
+    newOrigin[0] = oldOrigin[0];
+    newOrigin[1] = oldOrigin[1];
+    newOrigin[2] = oldOrigin[2];
+  }
+
+  // saveAsNewImage
+  {
+    int64 XP = ((xMax - xMin) + 1);
+    int64 YP = ((yMax - yMin) + 1);
+    int64 ZP = ((zMax - zMin) + 1);
+
+    std::vector<usize> tDims(3, 0);
+    tDims[0] = XP;
+    tDims[1] = YP;
+    tDims[2] = ZP;
+
+    auto spacing = srcImageGeom->getSpacing();
+    std::vector<float32> spacingVec(3);
+    for(usize i = 0; i < 3; i++)
+    {
+      spacingVec[i] = spacing[i];
+    }
+    auto action = std::make_unique<CreateImageGeometryAction>(destImagePath, tDims, newOrigin, spacingVec);
+    actions.actions.push_back(std::move(action));
+
+    auto newCellFeaturesPath = destImagePath.createChildPath(newCellFeaturesName);
+    {
+      auto action = std::make_unique<CreateDataGroupAction>(newCellFeaturesPath);
+      actions.actions.push_back(std::move(action));
+    }
+
+    auto* srcImageGeom = data.getDataAs<ImageGeom>(srcImagePath);
+    for(const auto& srcArrayPath : voxelArrayPaths)
+    {
+      auto* srcArray = data.getDataAs<IDataArray>(srcArrayPath);
+      if(srcArray == nullptr)
+      {
+        std::string ss = fmt::format("Could not find the DataArray at path '{}'", srcArrayPath.toString());
+        return {MakeErrorResult<OutputActions>(-5551, ss)};
+      }
+
+      auto numericType = static_cast<NumericType>(srcArray->getDataType());
+      auto numTuples = srcArray->getNumberOfTuples();
+      auto components = srcArray->getNumberOfComponents();
+      auto dataArrayPath = newCellFeaturesPath.createChildPath(srcArrayPath.getTargetName());
+      auto action = std::make_unique<CreateArrayAction>(numericType, std::vector<usize>{numTuples}, std::vector<usize>{components}, dataArrayPath);
+      actions.actions.push_back(std::move(action));
+    }
+  }
+
+  usize totalPoints = srcImageGeom->getNumberOfElements();
+
+  if(shouldRenumberFeatures)
+  {
+    std::vector<usize> cDims = {1};
```

Comment:
Is this meant to be used to check the number of components for the following array?